src/object/srv_obj.c

```diff
@@ -1585,18 +1582,29 @@ obj_local_rw_internal(crt_rpc_t *rpc, struct obj_io_context *ioc, daos_iod_t *io
 
 	if (obj_rpc_is_fetch(rpc) && !spec_fetch &&
 	    daos_csummer_initialized(ioc->ioc_coc->sc_csummer)) {
+		if (orw->orw_iod_array.oia_iods != iods) {
+			/* Need to copy iod sizes for checksums */
+			int i, j = 0;
+
+			for (i = 0; i < orw->orw_iod_array.oia_iod_nr; i++) {
+				if (skips != NULL && isset(skips, i))
```

Comment:
do we need to set orw->orw_iod_array.oia_iods[i].iod_size = 0 for this skip case?
we do zero it for the skip case in obj_set_reply_sizes()